exporter/exporter.collect.go

```
@@ -162,11 +162,10 @@ func (e *Exporter) CollectPrometheusMetric(ch chan<- semp.PrometheusMetric) {
 			} else {
 				ch <- e.semp.NewMetric(semp.MetricDesc["Global"]["up"], prometheus.GaugeValue, 0, "Unknown")
 			}
-			break
+		} else {
```

In case that multiple targets are given we have 2 use cases:

a.) it is a connection problem or something that effects all targets, this code is correct
b.) it is a problem that effects only a single target like: permissions problem or not available on hardware broker, ...   In this case i am not sure what is the best behavior. Because there will be metrics but only partial. In a fail fast strategy no metrics at all should be outputed.

---

Can you provide a simple ini file for me to test this behavior?
I tried to monitor two brokers with this ini (which is not working):
```
[solace]
listenAddr=0.0.0.0:9628
scrapeUri=URL1
...

[solace]
listenAddr=0.0.0.0:9629
scrapeUri=URL2
...
```

---

Target are not broker, in this case  i mean semp queries 
```ini
[endpoint.solace-det]
ClientStats = *|*
VpnStats = *|*
BridgeStats = *|*
QueueRates = *|*
QueueDetails = *|*
```

---

I understand your point now. From my perspective, `solace_up` indicates whether the Solace broker is reachable. This should not be associated with a broken configuration. Therefore, I would suggest setting `solace_up` to 1 when a single target cannot be reached.